src/CombFilter/CombFilterIf.cpp

```diff
@@ -0,0 +1,109 @@
+
+// standard headers
+
+// project headers
+#include "MUSI6106Config.h"
+
+#include "ErrorDef.h"
+#include "Util.h"
+
+#include "CombFilterIf.h"
+#include "RingBuffer.h"
+#include "CombFilter.h"
+
+static const char*  kCMyProjectBuildDate = __DATE__;
+
+
+CCombFilterIf::CCombFilterIf () :
+    m_bIsInitialized(false),
+    m_pCCombFilter(0),
+    m_fSampleRate(0)
+{
+    // this should never hurt
+    this->reset ();
+}
+
+
+CCombFilterIf::~CCombFilterIf ()
+{
+    this->reset ();
+}
+
+const int  CCombFilterIf::getVersion (const Version_t eVersionIdx)
+{
+    int iVersion = 0;
+
+    switch (eVersionIdx)
+    {
+    case kMajor:
+        iVersion    = MUSI6106_VERSION_MAJOR; 
+        break;
+    case kMinor:
+        iVersion    = MUSI6106_VERSION_MINOR; 
+        break;
+    case kPatch:
+        iVersion    = MUSI6106_VERSION_PATCH; 
+        break;
+    case kNumVersionInts:
+        iVersion    = -1;
+        break;
+    }
+
+    return iVersion;
+}
+const char*  CCombFilterIf::getBuildDate ()
+{
+    return kCMyProjectBuildDate;
+}
+
+Error_t CCombFilterIf::create (CCombFilterIf*& pCCombFilter)
+{
+    pCCombFilter = new CCombFilterIf ();
+    return Error_t::kNoError;
+}
+
+Error_t CCombFilterIf::destroy (CCombFilterIf*& pCCombFilter)
+{
+    delete pCCombFilter;
+    pCCombFilter = nullptr;
+    return Error_t::kNoError;
+}
+
+Error_t CCombFilterIf::init (CombFilterType_t eFilterType, float fMaxDelayLengthInS, float fSampleRateInHz, int iNumChannels)
+{
+    switch (eFilterType){
+        case CCombFilterIf::CombFilterType_t::kCombFIR:
+            m_pCCombFilter = new CCombFilterFIR (fMaxDelayLengthInS, fSampleRateInHz, iNumChannels);
```

Comment:
Is it possible to have a definition of the subclass where we don't set the parameter values directly through the constructor but rather use getters and setters?